cpr/session.cpp

```diff
@@ -119,9 +119,11 @@ void Session::Impl::SetHeader(const Header& header) {
                 header_string += ": " + item->second;
             }
             chunk = curl_slist_append(chunk, header_string.data());
```

Comment:
If `curl_slist_append` fails, then the return value is `NULL`. Let's store the value in a temporary and only assign `chunk` to that temporary when it's not `NULL`. That way we can be sure that `chunk` always points to the latest valid tail in the slist, and subsequent frees will not leak memory.